express-zod-api/src/documentation-helpers.ts

```diff
@@ -235,28 +265,34 @@ const getSupportedType = (value: unknown): SchemaObjectType | undefined => {
       : undefined;
 };
 
-export const depictEnum: Depicter = (
-  schema: z.ZodEnum<[string, ...string[]]> | z.ZodNativeEnum<z.EnumLike>,
-) => ({
-  type: getSupportedType(Object.values(schema.enum)[0]),
-  enum: Object.values(schema.enum),
+export const depictEnum: Depicter = ({ _zod: { def } }: $ZodEnum) => ({
+  type: getSupportedType(Object.values(def.entries)[0]),
+  enum: Object.values(def.entries),
 });
 
-export const depictLiteral: Depicter = ({ value }: z.ZodLiteral<unknown>) => ({
-  type: getSupportedType(value), // constructor allows z.Primitive only, but ZodLiteral does not have that constraint
-  const: value,
-});
+// @todo looks very similar to depictEnum, also takes values twice
+export const depictLiteral: Depicter = ({ _zod: { def } }: $ZodLiteral) => {
+  const values = Object.values(def.values);
+  if (values.length === 1)
+    return { type: getSupportedType(values[0]), const: values[0] };
+  return {
+    type: getSupportedType(values[0]),
+    enum: Object.values(def.values),
+  };
+};
 
 export const depictObject: Depicter = (
-  schema: z.ZodObject<z.ZodRawShape>,
+  schema: $ZodObject,
   { isResponse, next },
 ) => {
-  const keys = Object.keys(schema.shape);
-  const isOptionalProp = (prop: z.ZodTypeAny) =>
+  const keys = Object.keys(schema._zod.def.shape);
+  const isOptionalProp = (prop: $ZodType) =>
     isResponse && hasCoercion(prop)
       ? prop instanceof z.ZodOptional
-      : prop.isOptional();
-  const required = keys.filter((key) => !isOptionalProp(schema.shape[key]));
+      : (isResponse ? prop._zod.qout : prop._zod.qin) === "true";
```

Comment:
`qin` is not set on ZodOptional.
Filed a fix https://github.com/colinhacks/zod/pull/4153

Reply:
reverted to using `isOptional` so far